### DISCUSSION TITLE
Classifying With a Score - Conceptual Question

### DISCUSSION THREAD

**Discussion by natedailey7** [azavea/raster-vision#2097] [Q&A] [answered]
I'm trying to classify each pixel in an image with a score. That is, I'm trying to create a raster with pixels containing a percentage score. The application is to calculate the percentage canopy cover of a forest based on imagery, with a low score meaning little canopy cover and high score meaning dense canopy cover.

I'm thinking I could create discrete categories for different scores (0-0.1; 0.1-0.2, etc.) and then classify them with the semantic segmentation tools, but I'm wondering if you would recommend an approach that is more continuous. Is this in the realm of possibility for Raster Vision, or should I look elsewhere? Thanks in advance.

**Comment by AdeelH** [comment, marked as answer]
RV does have some support for regression but it is limited to chip-level regression i.e. predicting a single number for a full chip, rather for each pixel in the chip.

In your case, you could subclass the semantic segmentation learner and dataset classes into RasterRegression* classes with some  modifications. You would need to modify the loss function from cross-entropy to MSE or MAE, and modify the `train_step` and `valid_step` methods (see `RegressionLearner`), and also `postprocess_model_output` to not apply softmax. The `GeoDataset` class will have to be modified too.

So it will take some work, but if you do manage to get something working, I'd appreciate a PR!

**Reply by natedailey7** [reply]
Hello! I want to thank you for this response! I know this was a while ago, but I am still interested in this topic and wanted to ask a follow up question about how a "chip" is defined in Raster Vision terminology. Is the "chip-level regression" that you speak of just possible for chips as they are defined in the tutorial for [Working With Pre-Chipped Datasets](https://docs.rastervision.io/en/stable/usage/tutorials/prechipped_datasets.html)? That is, are chips defined as sets of non-georeferenced images (such as a set of PNGs)? Or is the definition of "chip" more broad than this?

**Reply by AdeelH** [reply]
By "chip" I just mean any window read from a larger geospatial scene. I was using "chip-level regression" to mean "image-level regression" to contrast it with "pixel-level regression". Whether you first save the chips as non-georeferenced images or read them directly from a GeoTIFF doesn't matter for the actual ML training.
